doc/volume-permissions-spec.md

```diff
@@ -0,0 +1,67 @@
+### Introduction
+
+Currently the Docker volume driver for VSphere allows any container on any VM with access to a given datastore
+to create, mount, and delete volumes at will. There are no limits to the size of the volumes that
+can be created or quotas for datastore usage for container volumes. This lack of access control
+takes the management of storage out of the hands of the IT administrator and puts it in the hands of
+the developer. This is a worrying proposition to many IT admins, and since they are the ones that
+must allow deployment of the driver and it's associated host agent, we must allow them to regain
+control with proper enforcement mechanisms that limit damage due to malfeasance or mistake.
+
+### Permissions Model
+
+Since docker containers run inside VMs, and can have root permissions, we cannot rely on the VM user
+for access control. Permissions must be granted to a VM, since VMs are created and managed
+by the IT administrator. This brings up a few issues:
+
```

Comment:
Suggest not linking up with VMs, instead allow users to be created and managed for storage entitlement. Each user is given a unique ID for the cluster (for example) and they can acquire storage capacity entitlements. Users provide their ID when allocating storage and the system manages the allocations per user.

The VM is purely a vehicle for running the user's containers and a way of plumbing backend VSphere storage to third party user containers.

The user can be running their containers on any number of VMs on any hosts in a cluster but they have a centrally managed entitlement to storage accessible via the cluster (see below on an alternate way of exposing storage to users abstracing out the storage architecture and instead positioning storage as classes based on capabilities).